packages/mwp-api-state/src/sync/index.js

```diff
@@ -88,8 +88,8 @@ export const getNavEpic = (routes, baseUrl) => {
 			resolvePrevQueries,
 		]).then(([newQueries, previousQueries]) => {
 			if (newQueries.filter(q => q).length === 0) {
-				// no valid queries - ignore
-				return [];
+				// no valid queries - jump straight to 'complete'
+				return [api.complete([])];
```

Comment:
this is sort of unrelated to the code optimization, but is useful regardless - it makes sure the server can render even if there are no queries matched in the route.